pyro/__init__.py

```diff
@@ -420,3 +422,17 @@ def _fn():
         # update_module_params must be True or the lifted module will not update local params
         return lifted_fn(name, nn_copy, update_module_params=True, *args, **kwargs)
     return _fn
+
+
+def set_validation(is_validate):
+    pyro.infer.set_validation(is_validate)
+
+
+@contextmanager
+def validate(is_validate):
```

Comment:
nit: maybe call this `validation_enabled` and default to `True`, this reads nicely
```py
with validation_enabled():
    # do stuff
```